.circleci/config.yml

```diff
@@ -165,113 +165,73 @@ jobs:
-          - &cache_key_beam
-            maven-repo-v1-beam-{{ .Branch }}-{{ checksum "../pom.xml" }}-{{ checksum "../ingestion-core/pom.xml" }}-{{ checksum "pom.xml" }}
-          - maven-repo-v1-beam-{{ .Branch }}-{{ checksum "../pom.xml" }}-{{ checksum "../ingestion-core/pom.xml" }}-
-          - maven-repo-v1-beam-{{ .Branch }}-{{ checksum "../pom.xml" }}-
-          - maven-repo-v1-beam-{{ .Branch }}-
-          - maven-repo-v1-beam-
-          - maven-repo-v1-
-    - &download_beam_resources
-      run:
-        name: Download resources needed for Beam tests
-        command: |
-          bin/download-schemas
-    - &maven_test
-      run:
-        name: Maven Test
-        command: |
-          mvn clean test $JAVA_COMPILER_RELEASE_FLAG
-    - &store_maven_test_results
-      store_test_results:
-        path: target/surefire-reports
-    - *report_code_coverage
-    - save_cache:
-        paths:
-          - ~/.m2
-        key: *cache_key_beam
-
-  ingestion-beam-integration:
-    <<: *ingestion_beam
-    steps:
-    # Checkout would fail if it were not the first step
-    # because checkout path would contain working dir
-    - *checkout
-    - &skip_forked_pr
-      run:
-        name: Early return if this build is from a forked PR
-        command: |
-          if [ -n "$CIRCLE_PR_NUMBER" ]; then
-            echo "Cannot pass creds to forked PRs, so marking this step successful"
-            circleci step halt
-          fi
-    - *skip_unmodified
-    - *restore_cache_beam
-    - *download_beam_resources
-    - &maven_integration_test
-      run:
-        name: Maven Integration Test
+          - &cache_key
+            maven-repo-v1-<< parameters.module >>-{{ .Branch }}-{{ checksum "../pom.xml" }}-{{ checksum "../ingestion-core/pom.xml" }}-{{ checksum "pom.xml" }}
+          - maven-repo-v1-<< parameters.module >>-{{ .Branch }}-{{ checksum "../pom.xml" }}-{{ checksum "../ingestion-core/pom.xml" }}-
+          - maven-repo-v1-<< parameters.module >>-{{ .Branch }}-{{ checksum "../pom.xml" }}-
+          - maven-repo-v1-<< parameters.module >>-{{ .Branch }}-
+          - maven-repo-v1-<< parameters.module >>-main-
+    - when:
+        condition: << parameters.integration >>
+        steps:
+          - run:
+              name: Early return if this build is from a forked PR
+              command: |
+                if [ -n "$CIRCLE_PR_NUMBER" ]; then
+                  echo "Cannot pass creds to forked PRs, so marking this step successful"
+                  circleci step halt
+                fi
+    - when:
+        condition:
+          equal: [ingestion-beam, << parameters.module >>]
+        steps:
+          - run:
+              name: Download resources needed for Beam tests
+              command: |
+                bin/download-schemas
+    - run:
+        name: Maven <<# parameters.integration >>Integration<</ parameters.integration >> Test
```

Comment:
This syntax of using boolean parameters for text blocks comes from https://circleci.com/docs/2.0/reusing-config/#boolean with `<<# ... >>` for true conditions and `<<^ ... >>` for false conditions.

Reply:
Worth adding this link as a code comment.